_publications/jors/paper.tex

```diff
@@ -0,0 +1,457 @@
+requirements whilst delegating back-end functionality to external
+services that provide dependable APIs~\cite{cmsfree}.
+
+The phase field method (PFM) describes material interfaces at the
+mesoscopic scale between atomic scale models and macroscale
+models~\cite{MOELANS2008268}. The PFM is well established and there
+are an assortment of code frameworks (\emph{e.g.}, FiPy~\cite{fipy},
+MMSP~\cite{mmsp}, MOOSE~\cite{moose}, PRISMS-PF~\cite{prisms-pf})
+available for solving the wide variety of phenomena associated with
+phase field (e.g. dendritic growth, spinodal decomposition, grain
+growth)~\cite{tonks}. However, it is difficult for novice as well as
+seasoned phase field practitioners to asses the capability of codes
+for different phenomena without extensive prototyping and
+groundwork. PFHub aims to provide a low barrier for comparing code
+output data using a standard set of metrics.
+
+PFHub is a community effort spearheaded by the Center for Hierarchical
+Materials Design at Northwestern University and the National Institute
+of Standards and Technology in support of phase-field code
+development. The current PFHub deployment~\cite{pfhub} focuses on
+improving cross-collaboration between phase-field code developers and
+practitioners by providing a standardized set of benchmark
+problems~\cite{bm1, bm2} and a workflow for uploading and comparing
+benchmark results from different phase-field codes.
+
+Community based scientific efforts often require web services to share
+and display data in unique ways between groups and institutions. These
+services are difficult to implement due to the groundwork required to
+investigate and prototype the many data-sharing and CMS tools
+available. The PFHub framework provides a template for other
+scientific projects beyond the phase field community. The method
+outlined in this paper of using static infrastructure coupled with
+small independent 3rd party web services providing a flexible approach
+eliminating the initial prototyping and on-going maintenance required
+for new infrastructure, while allowing developers to focus on their
+unique front-end data views.
+
+This paper presents the first deployment of the PFHub framework
+including its client-side focused design, how it employs external
+services and metadata about the code base. The paper describes the
+relative ease with which other scientific groups might adapt the
+framework for their own purposes and deploy using the fully
+reproducible Nix environment~\cite{nix}.
+
+\section*{Implementation and architecture}
+
+The PFHub framework provides a template for other small scientific
+communities to host custom content and integrate data from members of
+their community. The current deployment provides a facility for
+uploading, displaying and comparing results from benchmark problems
+supporting phase-field code developers and
+practitioners~\ref{fig:pfhub_frontpage}. However, the framework and
```

Comment:
Right now in the PDF the reference to the figure gets rendered as "and practitioners 1", so it's not clear the reader is being pointed to look at Figure 1- so unless there is something in the journal's formatting that will capture this, I'd suggest adding `see Figure ref{fig:pfhub_frontpage}`. Same with the references to Figure 3 and 4 below (Figure 2 looks OK)

Reply:
My mistake, thanks for noticing